File: src/Person.java
import java.util.Objects;

public abstract class Person implements Action{
    String name, status;
    public Person(String name, String status) {
        this.name = name;
        this.status = status;
    }
    @Override
    public String toString() {
        return "Person{" +
                "name='" + name + '\'' +
                ", status='" + status + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Person person = (Person) o;
        return name.equals(person.name) && Objects.equals(status, person.status);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, status);
    }

    public void action(String prevAct){}
}

File: kotlin_fans/Person.kt
import java.util.*

abstract class Person(var name: String, var status: String) : Action {
    override fun toString(): String {
        return "Person{" +
                "name='" + name + '\'' +
                ", status='" + status + '\'' +
                '}'
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val person = o as Person
        return name == person.name && status == person.status
    }

    override fun hashCode(): Int {
        return Objects.hash(name, status)
    }
}